pinot-core/src/main/java/org/apache/pinot/core/query/reduce/BrokerReduceService.java

```diff
@@ -168,6 +180,14 @@ public BrokerResponseNative reduceOnDataTable(@Nonnull BrokerRequest brokerReque
     brokerResponseNative.setNumSegmentsMatched(numSegmentsMatched);
     brokerResponseNative.setTotalDocs(numTotalRawDocs);
     brokerResponseNative.setNumGroupsLimitReached(numGroupsLimitReached);
+    if (numConsumingQueried > 0) {
+      if (minConsumingIndexTs == Long.MAX_VALUE) {
+        LOGGER.error("Invalid lastIndexedTimestamp across {} consuming segments", numConsumingQueried);
```

Comment:
Log the query ID here so that we can use it to correlate query logs. Also, please ensure that if this condition happens we override the throttle in the broker logs.

Reply:
Can you add the query ID here?

Reply:
@mcvsubbu Seems we already lost the requestId info in this layer. I think it should be fine because it should not happen, or any query can hit this error message.